src/GirTool/GenerateCommand.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System.CommandLine;
+using System.CommandLine.Binding;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Generator3;
+using GirLoader;
+
+namespace GirTool;
+
+public class GenerateCommand : Command
+{
+    public GenerateCommand() : base(
+        name: "generate",
+        description: "Generate C# bindings from gir files"
+    )
+    {
+        AddArgument(new Argument<string[]>(
+            name: "input",
+            description: "The gir files which should be processed"
+        ));
+
+        AddOption(new Option<string>(
+            aliases: new[] { "-o", "--output" },
+            description: "The directory to write the generated C# files to",
+            getDefaultValue: () => "./Libs"
+        ));
+
+        AddOption(new Option<string>(
+            aliases: new[] { "-s", "--search-path" },
+            description: "The directory which is searched vor dependent gir files",
+            getDefaultValue: () => "./Gir"
+        ));
+
+        AddOption(new Option<bool>(
+            aliases: new[] { "-d", "--disable-async" },
+            getDefaultValue: () => false,
+            description: "Generate files synchronously"
```

Comment:
Maybe mention how this is useful for debugging or if something goes wrong? Otherwise it doesn't really explain the purpose of the option imo. Not a big deal however.

e.g. "Generate files synchronously - useful for debugging"